edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeAgentDockerModule.cs

```diff
@@ -21,9 +23,24 @@ public override bool Equals(IModule<DockerConfig> other)
                 return false;
             if (ReferenceEquals(this, other))
                 return true;
-            return string.Equals(this.Name, other.Name) &&
-                string.Equals(this.Type, other.Type) &&
-                string.Equals(this.Config.Image, other.Config.Image);
+
+            if (!string.Equals(this.Name, other.Name) ||
+                !string.Equals(this.Type, other.Type) ||
+                !string.Equals(this.Config.Image, other.Config.Image))
+                return false;
+
+            IDictionary<string, string> labels = other.Config.CreateOptions?.Labels ?? new Dictionary<string, string>();
+            if (!labels.TryGetValue(Core.Constants.Labels.CreateOptions, out string createOptions) ||
+                !labels.TryGetValue(Core.Constants.Labels.Env, out string env))
+                return true;
+
+            // If the 'net.azure-devices.edge.create-options' and 'net.azure-devices.edge.env' labels exist
+            // on the other IModule, compare them to this IModule
+            if (JsonConvert.SerializeObject(this.Config.CreateOptions) != createOptions)
```

Comment:
```suggestion
            if (!string.Equals(JsonConvert.SerializeObject(this.Config.CreateOptions), createOptions))
```

Reply:
It makes sense to do this so I can add the `StringComparison.OrdinalIgnoreCase` parameter.

Reply:
Actually I don't want to ignore case, so just `StringComparison.Ordinal`. That's equivalent to what I have now, but I'll change it to be explicit.

Reply:
Done